### PR TITLE
gtk-sharp 2.12.42 (new formula)

### PR DESCRIPTION
**Pull request by DoomHammer**
- [X] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/gtk-sharp.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+class GtkSharp < Formula
+  desc "Graphical User Interface Toolkit for mono and .Net"
+  homepage "http://www.mono-project.com/GtkSharp"
+  url "http://download.mono-project.com/sources/gtk-sharp212/gtk-sharp-2.12.42.tar.gz"
+  sha256 "f3b009bb73e3251378063b6f09786609cd4c061f3f8bf552f0ea663245c045c9"
```
Comment (sjackman) [inline Formula/gtk-sharp.rb]: Please add `# tag "linuxbrew"`

**Comment by sjackman** [comment]
I'm thinking that these non-critical formula unique to Linuxbrew should go in a new tap.
@Linuxbrew/core Your thoughts, and if so, what should it be named?
I'm thinking `linuxbrew/homebrew-linux`

**Comment by rwhogg** [comment]
> I'm thinking that these non-critical formula unique to Linuxbrew should go in a new tap.
@Linuxbrew/core Your thoughts, and if so, what should it be named?

My concern is it's one more tap that requires Travis configuration, bottle setup, etc. Also, someone would have to maintain it. Doing that is already challenging enough for dupes and xorg.

On the bright side, it's clearer to contributors that they would be able to change things in such a tap much more freely than they currently can in homebrew-core, which in turn would make it easier for non-maintainers to help us maintain it. And useful formulae that don't work (or don't work well) on macOS certainly deserve a home of some sort.

> I'm thinking linuxbrew/homebrew-linux

There's a possibility people might take that to mean it's for installing the kernel itself. We could call it linuxbrew/homebrew-extra, emphasizing that it's an add-on to core and that you don't need it unless you don't want to.

**Comment by sjackman** [comment]
> My concern is it's one more tap that requires Travis configuration, bottle setup, etc. Also, someone would have to maintain it. Doing that is already challenging enough for dupes and xorg.

Yep, that's true. It could be an opportunity to document what's needed to set up a new tap. There is [`dev-cmd/tap-new.rb`](https://github.com/Linuxbrew/brew/blob/master/Library/Homebrew/dev-cmd/tap-new.rb), which I've never looked at. We could copy that to `Linuxbrew/developer` and name it `tap-new-linuxbrew` and adapt it for Linuxbrew.

Would any of @rwhogg, @DoomHammer and/or @maxim-belkin want to volunteer to maintain this new tap?

The criterion for whether a formula goes in Linuxbrew/core or Linuxbrew/extra would be
Does a formula in Linuxbrew/core depend on this formula?
Yes -> It goes in Linuxbrew/core
No -> It goes in Linuxbrew/extra

> We could call it linuxbrew/homebrew-extra, emphasizing that it's an add-on to core and that you don't need it unless you don't want to.

I'm happy with the name `Linuxbrew/extra`.

**Comment by DoomHammer** [comment]
This particular formula _may_ be useful to upstream as well, I think. I would suggest splitting the conversation about new tap to a separate issue.

**Comment by maxim-belkin** [comment]
ping @DoomHammer 
You can either submit this upstream or to Linuxbrew/extra

**Comment by DoomHammer** [comment]
Guess I will try one first and if it fails the other :)

**Pull request closed without merging**